### PR TITLE
Add additional logging to find race condition

### PR DESCRIPTION
**Pull request by demophoon**
It is possible for a job to disappear and have the event handler attempt to use job before checking if there was an error on the error stream which can cause Waypoint to panic.

### CODE REVIEW

File: pkg/server/singleprocess/service_runner.go [hunk 1]
```diff
@@ -881,6 +888,10 @@ func (s *Service) handleJobStreamRequest(
 	log.Trace("event received", "event", req.Event)
 	switch event := req.Event.(type) {
 	case *pb.RunnerJobStreamRequest_Complete_:
+		if event.Complete == nil {
+			log.Error("event.Complete was not filled when we expected it to be", "id", job.id)
```
Comment (briancain) [inline pkg/server/singleprocess/service_runner.go:892]: Maybe log the `req.Event` too?

File: pkg/server/singleprocess/service_runner.go [hunk 2]
```diff
@@ -881,6 +888,10 @@ func (s *Service) handleJobStreamRequest(
...
+			return hcerr.Externalize(log, err, "failed to complete job", "id", job.id)
```
Comment (briancain) [inline pkg/server/singleprocess/service_runner.go:893]: I think the linter is mad about this one? There's no err available on this branch I think.

File: pkg/server/singleprocess/service_runner.go (hunk 1 above)
Comment (demophoon) [reply pkg/server/singleprocess/service_runner.go]: Added!

File: pkg/server/singleprocess/service_runner.go (hunk 2 above)
Comment (demophoon) [reply pkg/server/singleprocess/service_runner.go]: Fixed, I think my vim plugin stopped working and didn't catch that `job.id` should actually be `job.Id` 😨

**Review by briancain — APPROVED** [review]

**Pull request merged**